src/main.rs

```diff
@@ -166,18 +166,43 @@ fn main() {
         libc::signal(SIGINT, mem::transmute(handle_sigint));
     }
 
-    let format = |record: &LogRecord| {
-        let t = time::now();
-        format!("{}.{:03}: {}{:5} {}",
-            time::strftime("%Y-%m-%d %H:%M:%S", &t).unwrap(),
-            t.tm_nsec / 1000_000,
-            tid_str(),
-            record.level(),
-            record.args()
-        )
-    };
     let mut builder = LogBuilder::new();
-    builder.format(format).filter(None, LogLevelFilter::Info);
+    let istty = unsafe { libc::isatty(libc::STDERR_FILENO as i32) } != 0;
+    if istty {
+        // Colorized output formatter
+        let format = |record: &LogRecord| {
+            let t = time::now();
+            let level_color = match record.level() {
+                log::LogLevel::Error => "\x1b[1;31m",  // bold red
+                log::LogLevel::Warn  => "\x1b[1;33m",  // bold yellow
+                log::LogLevel::Info  => "\x1b[1;32m",  // bold green
+                log::LogLevel::Debug => "\x1b[1;34m",  // bold blue
+                log::LogLevel::Trace => "\x1b[1;35m"   // bold magenta
```

Comment:
Line 182 resets the color with **\x1b[0m** right after the log level keyword is printed, so that keyword is the only part of the line that is color coded. The color is unable to bleed over into *println!*s.

We can still add color themes later if you don't like the green. ;)